english for core classes with methods, attributes and (most) comments

git0some opened the pull request.

git0some committed:
Core Elements English now

git0some committed:
base and core classes attributes and methodes in English. and some comments

git0some committed:
Classes, methods, attributes in english and most of the comments

tomwendel commented:
Wohooo. How cool is that?!

tomwendel commented on SimulationCore/Simulation/PlayerAnalysis.cs:
```diff
@@ -444,7 +444,7 @@ private List<PlayerInfo> analyseAssembly(ModuleDefinition module, bool checkRule
                 }
 
-                #region Erkennung �lterer KI-Versionen
+                #region Erkennung aelterer KI-Versionen
```
;)

tomwendel commented on SimulationCore/Simulation/CoreColony.cs:
```diff
@@ -39,15 +39,15 @@ internal sealed class CoreColony
-        internal int H�heI2;
+        internal int WidthI;
+        internal int WidthI2;
+        internal int HightI;
```
This should be "height" I guess

tomwendel commented on SimulationCore/Simulation/CoreColony.cs:
```diff
@@ -73,32 +73,32 @@ internal sealed class CoreColony
         /// Die maximalen Lastwerte aller Kasten des Volkes.
         /// </summary>
-        internal readonly int[] Last;
+        internal readonly int[] Burden;
```
We usually translate "Last" with "Load" all over the place. To be consistent I would suggest to use "Load" also here.

tomwendel commented on SimulationCore/Deutsch/Insekt.cs:
```diff
@@ -20,79 +20,79 @@ public override int Id
         /// Gibt die Angriffsstärke an
         /// </summary>
-        public int Angriff
+        public int Angriffstaerke
```
This one will break the API of the Ant. Here we need to stick with "Angriff"

git0some replied:
Danke für die Rückmeldung.
ich bin auch kein Native Speaker aber ich bin beim Compilieren mit Rider auf einige Probleme mit den Umlauten gestoßen, daher der Versuch die Simulation-Classes auf Englisch gerade zu ziehen.

Eine Frage auf die ich beim übersetzten gestoßen bin: 
ist Range für Movement Range oder für Smell Range oder für beides?
CasteInfo.cs Zeile 102
            // Prüfen, ob der Riechweitemodifikator im Rahmen ist
            if (Range < SimulationSettings.Custom.CasteSettings.MinIndex ||
                Range > SimulationSettings.Custom.CasteSettings.MaxIndex)
in allen anderen Zusammenhängen, die ich bisher gesehen habe war Range die Bewegungsreichweite und nicht die Riechreichweite
Update obiger Files folgt

tomwendel replied:
Bei dem ganzen deutschen Quellcode und den Kommentaren brennen mir inzwischen ja auch die Augen :D Deshalb bin ich so froh, dass du dich dieser Sache annimmst.

Was die Range angeht: Es gibt im Grunde ja keine Smell Range. Die Ameise riecht eine Markierung, sobald deren Radius den Kollisionsradius der Ameise kreuzt. Wie weit die Ameise also riechen kann, ist nur von der Markierung abhängig. Daher handelt es sich hier um die maximale Laufreichweite der Ameise.

tomwendel reviewed (changes requested):
Wow. This looks quite awesome. Thank you for the work.

I have only two notes:
- I found a place where the actual ant API was changed (Angriff -> Angriffsstaerke) which potentially breaks existing ant implementation
- There was a translation for "Last" by "Burdon". but usually we translate this with "Last". I think it's better to be consistent with all the other translations here

You even corrected the wiki links <3

git0some committed:
update to  english for core classes with methods, attributes and (most) comments

al3xander committed:
work in progress (Deutsch->english)
litte update to ColorFinder.cs division by two for each RGB value in operator + and check for 0 in operator /

al3xander committed:
work in progress (Deutsch->english)

al3xander committed:
work in progress (Deutsch->english)

al3xander committed:
Merge remote-tracking branch 'origin/main'

git0some committed:
some more (Deutsch->english) and Correction for making a new colony with english baseAnt because of
AntMe.Simulation.RuleViolationException: Ameisen der Version 1.1 oder kleiner werden in dieser Version nicht mehr unterstützt.
   bei AntMe.Simulation.PlayerAnalysis.analyseAssembly(ModuleDefinition module, Boolean checkRules)

al3xander committed:
last step in the Deutsch->English translation for all not german classes

al3xander committed:
found grün in commented source ModelManager.cs

al3xander committed:
more german textes to translate found, some clean up regarding first character uppercase and punctuation marks

al3xander committed:
Update VideoRecorderControl.cs (#1)

tomwendel commented:
Huch. Hier ist ja noch richtig was passiert. Lass mich gerne wissen, wenn ihr soweit fertig seid und ich noch mal nen Review machen kann?

al3xander committed:
Several changes in SimulationCore English Deutsch and Simulation. To make it clear where the ant/bug/insect attributes and methods are coming from.
Instead of ...Base  in most cases ...CoreInsect.

al3xander committed:
zustand -> state in SharedComponents\AntVideo\Block files

git0some commented:
> Huch. Hier ist ja noch richtig was passiert. Lass mich gerne wissen, wenn ihr soweit fertig seid und ich noch mal nen Review machen kann?

Ja, ich glaube jetzt ein ganz guter Stand erreicht zum Reviewen. 
Ein paar Vereinfachungen im Code  habe ich vorgenommen und eine programmtechnische Ergänzung laut ursprünglichem Kommentar in SharedComponents\Tools\ColorFinder.cs  // Division by two for each RGB value.

al3xander committed:
English ants ran into a problem with initial amount=0 for fruits and sugar hills. SimulationSettings.Custom -> SimulationSettings.Default is a workaround.

al3xander committed:
update in AntInfoBox constructor parameter description and little adaption in BaseAnt

al3xander committed:
adding additional functionality to Insect.cs according to german Insekt.cs

al3xander committed:
Typo in currentEngergyCoreInsect => currentEnergyCoreInsect

al3xander committed:
CasteIndexCoreAnt broke the logic for chosing caste for new ants. Back to CasteIndexCoreInsect again.

al3xander committed:
code thightening

al3xander committed:
added AreaChanged check for null

al3xander committed:
rearrange rotation speed and speed to fit to german version
